blogposts/lsif-indexer.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Writing an LSIF Indexer"
+author: Uwe Hoffmann
+publishDate: 2019-09-28T10:00-07:00
+tags: [
+  blog
+]
+slug: writing-an-lsif-indexer
+heroImage: https://about.sourcegraph.com/sourcegraph-mark.png
+published: true
+---
+
+At a high level, an LSIF indexer analyzes a collection of input source code files and produces an LSIF data file. This LSIF data file, in turn, is used by an editor or a Sourcegraph instance or some other developer tool to offer functionality like "Jump to Definition", "Show References", hover results with documentation and type information, semantic search, etc.
+
+This article explores what it would take to write an LSIF indexer. Along the way, we will learn a little bit about parsing, walking abstract syntax trees, scopes and the LSIF file format.
+
+To make things concrete for our exploration, we will write an LSIF indexer for the programming language Jsonnet, and we will limit ourselves to providing enough data for "Jump to Definition" in a Jsonnet file including its imports.
```

Comment:
@dadlerj have we standardized on "go to definition" vs "jump to definition"?

Reply:
Good question, had to look this up — looks like "go to" is preferred https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/doc/dev/style_guide.md#specific-terms-in-prose

It has more usages too: https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+"go+to+definition" vs. https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/sourcegraph%24+"jump+to+definition"

@vanesa you cool with that?

Reply:
Agree!

Reply:
done